trl/trainer/grpo_trainer.py

```diff
@@ -453,10 +642,18 @@ def __init__(
             model = self._enable_gradient_checkpointing(model, args)
 
         # Processing class
+        self.is_vision_model = model.config.model_type in MODEL_FOR_VISION_2_SEQ_MAPPING_NAMES.keys()
         if processing_class is None:
-            processing_class = AutoTokenizer.from_pretrained(model.config._name_or_path, padding_side="left")
-        if processing_class.pad_token is None:
-            processing_class.pad_token = processing_class.eos_token
+            if self.is_vision_model:
+                processing_class = AutoProcessor.from_pretrained(model.config._name_or_path, padding_side="left")
+            else:
+                processing_class = AutoTokenizer.from_pretrained(model.config._name_or_path, padding_side="left")
+
```

Comment:
duplicated line

Reply:
no one is `AutoProcessor` and the other `AutoTokenizer`